packages/core/src/deploy.ts

```diff
@@ -63,6 +64,8 @@ export async function deploy<
   defaultSender?: string;
   strategy: DeploymentStrategyType;
 }): Promise<DeploymentResult> {
+  const executionStrategy: ExecutionStrategy = strategy as ExecutionStrategy;
```

Comment:
What's the purpose of this type cast?

Reply:
The `DeploymentStrategyType` is the reduced and simplified type we are currently exposing to the public. `ExecutionStrategy` is the full internally used type we don't want to expose, so as soon as we are behind the public api we convert from reduced to full for internal use.